requirements/quality-extra.txt

```diff
@@ -0,0 +1,2 @@
+pylint
+edx-lint                  # edX pylint rules and plugins
```

Comment:
quality.txt = pip-compile quality.in + quality-extra.txt.  

So those requirements are in quality.txt *because* they are in quality-extra.txt.  The reason for including them there instead of in quality.in is because pip-compile comes up against the pip-tools bug mentioned in the Makefile.

See the specification for `make upgrade` in Makefile, as well as the docs for the example workflow in [pip-tools](https://github.com/nvie/pip-tools#example-usage-for-pip-compile) for the relationship between dev.in, dev-extra.txt and dev.txt.